### ISSUE TITLE
Adaptive weights of GradNorm

### ISSUE THREAD

**Issue by S-Beers** [VICO-UoE/KD4MTL#3]
Hi, 
I have a question about the implementation of GradNorm in MTAN.  If I understand correctly, the adaptive weights are initialized at every batch. `w = torch.ones(len(tasks)).float().cuda()` . However, this differs from the paper, where the weights are only initialized once. I hope you can shed some light on this issue.

**Comment by WeihongLi-ac** [comment]
Hi,

Thanks for the question. For GradNorm, we update the loss weights using `Weights = Weight(tasks).cuda()` (line 61 for MTAN) and pass them to `w` (line 147-165) so loss weights are initialized at the beginning of the whole training rather than at every iteration. As we implemented different loss weighting methods within a single script , we will pass the loss weights values computed by the chosen method to `w` or `w` will be one for each task for the uniform loss weighting baseline. 

If you are also interested in more loss weighting strategies or more implementation of learning multi-task/domain models, please refer to our updated version: [Universal Representations](https://github.com/VICO-UoE/UniversalRepresentations).

Hope this will help!
Thanks!
